### ISSUE TITLE
How to compress dist/ chunks using express compression. 

### ISSUE THREAD

**Issue by seabass118** [szymmis/vite-express#47]
I want to send the chunks to the client using GZIP. However it does not seem to work when using vite-express.

Server code:
```import express from "express";
import ViteExpress from "vite-express";
import compression from "compression";

const app = express();

app.use(compression());

ViteExpress.config({ mode: "production" })

ViteExpress.listen(app, 3000, () => console.log("Server is listening..."));
```

Chunks:
![image](https://user-images.githubusercontent.com/63876056/236829287-32d6b28c-33d6-477f-a909-d99202ef7257.png)

![image](https://user-images.githubusercontent.com/63876056/236829950-e6dd17ec-e939-45c4-a880-136f43aed2af.png)

**Comment by szymmis** [comment]
Yeah, I think that I know exactly where the problem is. `vite-express` internally uses `express.static` middleware, and it injects it before any middleware that you as a developer set. So in that case this `express.static` middleware is first in order before your compression middleware so that's why it doesn't compress how you want. What is problematic is to figure out how to tackle that problem to both fix it and also make as small changes to the library as possible so that we don't alter the current usage.

**Comment by seabass118** [comment]
> Yeah, I think that I know exactly where the problem is. `vite-express` internally uses `express.static` middleware, and it injects it before any middleware that you as a developer set. So in that case this `express.static` middleware is first in order before your compression middleware so that's why it doesn't compress how you want. What is problematic is to figure out how to tackle that problem to both fix it and also make as small changes to the library as possible so that we don't alter the current usage.

Thats what I was thinking. My initial thought was to just try chunk splitting to try and reduce the bundle size but I am unable to access the 'vite-express' module  in the build context since it runs/builds as middleware for the app. So pretty lost as to what to do to reduce bundle sizes.

**Comment by seabass118** [comment]
You could maybe add a new parameter in ```ViteExpress.config({ ... })``` where we can pass through a compression type for production deployments.

**Comment by seabass118** [comment]
> That would work, but it is a fix for a specific case, and I don't want to add configuration option for every case because you know - the more code you write, the more bugs you'll produce 😄.
> 
> What I want to do is: make `vite-express` inject it's static middleware after all of your middleware, so right between your last middleware and the first registered route. That way it will be possible to control what is happening before `vite-express` sends static files in requests. By doing that, you won't need to add any change to your code to make it work. The only case where it might cause problems is when you have some kind of authentication/authorization middleware that make block static files if we flip the order with the changes I proposed.
> 
> I can we can overcome by introducing an export of its custom static middleware from `vite-express` something like `ViteExpress.static()` for you to use when you need to control when the static files middleware is set up.
> 
> So imagine this situation:
> 
> ```js
> import express from "express"
> import compression from "compression"
> import someAuthMiddleware from "some/path"
> 
> const app = express()
> 
> app.use(compression())
> app.use(ViteExpress.static())
> app.use(someAuthMiddleware())
> 
> ViteExpress.listen(app, 3000,  () => /*...*/)
> ```
> 
> But this is only when you have some kind of middleware that blocks requests and it block static files used for let's say displaying login page.
> 
> In your case the code would be the same as you wrote:
> 
> ```js
> import express from "express"
> import compression from "compression"
> 
> const app = express()
> 
> app.use(compression())
> 
> ViteExpress.listen(app, 3000,  () => /*...*/)
> ```
> 
> What do you think about this? I'd love to hear your opinion, and if anything is unclear let me know, and I'll try to explain it better.

That looks good. Injecting middle-ware before ```vite-express``` makes a lot more sense for production grade applications.  Please keep me updated.

**Comment by szymmis** [comment]
Hi @seabass118, sorry that it took so long but I struggled to find free time to think about that and do necessary changes. I've published v0.7.0 just now, I'm not entirely sure that I didn't break anything but let's keep our fingers crossed. You should be able to use the code that I've wrote above. You can also check the README [section](https://github.com/szymmis/vite-express#static--requesthandler), and if something is still unclear or doesn't work, please let me know. Thanks a lot for raising this issue!
